core/libp2p/crypto/key_generator/key_generator_impl.cpp

```diff
@@ -37,11 +37,11 @@ namespace libp2p::crypto {
     template <class KeyStructure, class Function>
     outcome::result<std::vector<uint8_t>> encodeKeyDer(KeyStructure *ks,
                                                        Function *function) {
-      gsl::owner<unsigned char *> buffer = nullptr;
-      // NOLINTNEXTLINE(cppcoreguidelines-owning-memory)
-      auto cleanup = gsl::finally([buffer]() {
-        // NOLINTNEXTLINE(cppcoreguidelines-no-malloc, hicpp-no-malloc)
-        OPENSSL_free(buffer);
+      unsigned char *buffer = nullptr;
+      auto cleanup = gsl::finally([pptr = &buffer]() {
+        if (*pptr != nullptr) {
+          OPENSSL_free(*pptr);
+        }
```

Comment:
what is the point of passing pointer to pointer in cleanup?

Reply:
lambda captures a copy of pointer when it is nullptr, so it will not be freed